### ISSUE TITLE
Chromium mac-494755 crashes on OS X 10.12.6

### ISSUE THREAD

**Issue by hparadiz** [puppeteer/puppeteer#479]
I'm experiencing tab crashes while writing some automated screenshot scripts using Puppeteer.

I've attached three different crash dumps in hopes that it will be useful to the team in the future.

The page which is crashing has a lot of canvas drawn [Highcharts](https://www.highcharts.com/) with a high volume of data being plotted.

The page in question loads just fine in Google Chrome Version 61.0.3163.49 (Official Build) beta (64-bit)

[Chromium Crash Dumps.zip](https://github.com/GoogleChrome/puppeteer/files/1243243/Chromium.Crash.Dumps.zip)

**Comment by ebidel** [comment]
Can you post the code that repros the issue?

**Comment by hparadiz** [comment]
```js
        await global.browser.goto(url);
        await global.browser.waitForSelector('input[name="login"]');

        await global.browser.setInput('input[name="login"]',process.env.IMAGINGUSER);
        await global.browser.setInput('input[name="password"]',process.env.IMAGINGPW);   
        await global.browser.click('[name="commit"]');

        // crashes here
```

I'm confident this bug is not really a bug with Puppeteer but with the Chromium build.

Sadly I can't link to the URL as it lives on the internal corp network but here's a sanitized screenshot to give you an idea of just how insanely large the data set for these charts is.
![image](https://user-images.githubusercontent.com/195216/29588703-86b0418e-8760-11e7-93c1-0b68f8761d2d.png)

**Comment by ebidel** [comment]
One thing you could try is install from master and see if that fixes the canvas issue. We _just_ rolled Chromium to a later version (r496140): https://github.com/GoogleChrome/puppeteer/pull/465.

`npm i --save https://github.com/GoogleChrome/puppeteer`

**Comment by hparadiz** [comment]
Sounds good! I'll try that tomorrow morning and let you know.

**Comment by hparadiz** [comment]
So I updated to r496140 and I kept getting the same issue.

On a whim I tried this instead and it worked:
```js
        await global.browser.goto(url);
        //await global.browser.waitForSelector('input[name="login"]');
        await global.browser.wait(1000);
        await global.browser.setInput('input[name="login"]',process.env.IMAGINGUSER);
        await global.browser.setInput('input[name="password"]',process.env.IMAGINGPW);   
        await global.browser.click('[name="commit"]');
```

So I'm thinking there's some issue with waitForSelector which causes a tab to crash.

*Update*
- Still crashes most of the time. Occasionally loads just fine. Trying to nail down the cause.

**Comment by aslushnikov** [comment]
@hparadiz would you be able to provide a repro scenario? Unfortunately, this is not actionable unless we have a way to reproduce this.

**Comment by hparadiz** [comment]
@aslushnikov 
I tried to build you guys a demo where that would happen but unfortunately (or perhaps fortunately) none of my offline local demos experienced the crash. Even when I exported and used the same data. I think it may have had something to do with network transfer and lag.

But good news! The tab no longer crashes!
After 74c53f37588a15b8bc04ce752c53a662949b44da I stopped experiencing this issue.

I think we can close this.
